src/cli/Peers.ts

```diff
@@ -0,0 +1,387 @@
+        const publicKey = fs.readFileSync(publicKeyPath).toString();
+
+        const successful = await client.pingPeer(nodePath, publicKey);
+
+        if (successful) {
+          pkLogger('peer successfully pinged', PKMessageType.SUCCESS);
+        } else {
+          pkLogger('ping timed out', PKMessageType.WARNING);
+        }
+      }),
+    );
+}
+
+function makeFindPeerCommand() {
+  return new commander.Command('find')
+    .description('find a peer based on a public key')
+    .option('--node-path <nodePath>', 'node path')
+    .requiredOption('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+        const publicKeyPath = options.publicKey;
+        // read in publicKey if it exists
+        const publicKey = fs.readFileSync(publicKeyPath).toString();
+
+        const successful = await client.findPeer(nodePath, publicKey);
+
+        if (successful) {
+          pkLogger('peer successfully pinged', PKMessageType.SUCCESS);
+        } else {
+          pkLogger('ping timed out', PKMessageType.WARNING);
+        }
+      }),
+    );
+}
+
+function makeFindSocialPeerCommand() {
+  return new commander.Command('social')
+    .description('find a peer based on a handle and service, e.g. john.smith and github')
+    .option('--node-path <nodePath>', 'node path')
+    .requiredOption('-h, --handle <handle>', 'handle of the user on the specified service, e.g. john.smith')
+    .requiredOption('-s, --service <service>', 'service where the handle can be found, e.g. github')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+        const handle = options.handle;
+        const service = options.service;
+
+        const successful = await client.findSocialPeer(nodePath, handle, service);
+
+        if (successful) {
+          pkLogger('peer successfully pinged', PKMessageType.SUCCESS);
+        } else {
+          pkLogger('ping timed out', PKMessageType.WARNING);
+        }
+      }),
+    );
+}
+
+function makeStealthCommand() {
+  return new commander.Command('stealth')
+    .description('toggle stealth mode on or off')
+    .option('--node-path <nodePath>', 'node path')
+    .option('--active', 'toggle stealth mode to active')
+    .option('--no-active', 'toggle stealth mode to not active')
```

Comment:
I suggest separate sub commands for this. Not options. Because this is the end result, and not about parameterizing some other action.

Reply:
we can have `pk peers stealth active` and `pk peers stealth inactive`? I think that is what you mean

Reply:
done.